workflows_api/runtime/src/collection_publisher.py

```diff
@@ -137,14 +140,33 @@ def create_cog_collection(self, dataset: COGDataset) -> dict:
                 },
             }
         )
-        collection_stac["item_assets"] = {
-            "cog_default": {
-                "type": "image/tiff; application=geotiff; profile=cloud-optimized",
-                "roles": ["data", "layer"],
-                "title": "Default COG Layer",
-                "description": "Cloud optimized default layer to display on map",
+        collection_stac["item_assets"] = {}
+
+        dataset_dict = dataset.dict(exclude_unset=True)
+
+        discovery_items_assets = []
+        if (dataset_dict.get("discovery_items")):
+            discovery_items_assets = [discovery_item.get("assets") for discovery_item in dataset_dict.get("discovery_items") if discovery_item.get("assets") is not None]
+
+        if dataset_dict.get("item_assets"):
+            collection_stac["item_assets"] = dataset_dict.get("item_assets")
+        elif discovery_items_assets:
+            for discovery_asset in discovery_items_assets:
+                for key, asset in discovery_asset.items():
+                    collection_stac["item_assets"][key] = {
+                        k: v for k, v in asset.items() if k != "regex"
+                    }
+        else:
+            collection_stac["item_assets"] = {
+                "cog_default": {
+                    "type": "image/tiff; application=geotiff; profile=cloud-optimized",
+                    "roles": ["data", "layer"],
+                    "title": "Default COG Layer",
+                    "description": "Cloud optimized default layer to display on map",
+                    "regex": "*"
```

Comment:
actually, this should probably be whatever `filename_regex` is instead

Reply:
Updated with the mentioned change:
![image](https://github.com/NASA-IMPACT/veda-data-airflow/assets/38744044/d3e934f7-6ce5-464d-a17f-6e87486a9a50)